src/main/java/software/amazon/awssdk/crt/eventstream/Header.java

```diff
@@ -0,0 +1,372 @@
+                header.headerValue = new byte[2];
+                buffer.get(header.headerValue);
+                break;
+            case 4:
+                header.headerType = HeaderType.Int32;
+                header.headerValue = new byte[4];
+                buffer.get(header.headerValue);
+                break;
+            case 5:
+                header.headerType = HeaderType.Int64;
+                header.headerValue = new byte[8];
+                buffer.get(header.headerValue);
+                break;
+            case 6:
+                short bufLen = buffer.getShort();
+                byte[] bufValue = new byte[bufLen];
+                buffer.get(bufValue);
+                header.setValue(bufValue);
+                break;
+            case 7:
+                short strLen = buffer.getShort();
+                byte[] strValue = new byte[strLen];
+                buffer.get(strValue);
+                header.setValue(new String(strValue, StandardCharsets.UTF_8));
+                break;
+            case 8:
+                header.headerType = HeaderType.TimeStamp;
+                header.headerValue = new byte[8];
+                buffer.get(header.headerValue);
+                break;
+            case 9:
+                header.headerType = HeaderType.UUID;
+                header.headerValue = new byte[16];
+                buffer.get(header.headerValue);
+                break;
+            default:
+                throw new CrtRuntimeException("Invalid event-stream header buffer.");
+        }
+
+        return header;
+    }
+
+    public void writeToByteBuffer(ByteBuffer buffer) {
+        buffer.put((byte)headerName.length());
+        buffer.put(headerName.getBytes(StandardCharsets.UTF_8));
+        buffer.put((byte)headerType.getEnumIntValue());
+
+        if (headerType != HeaderType.BooleanFalse && headerType != HeaderType.BooleanTrue) {
+            buffer.put(headerValue);
+        }
+    }
+
+    public String getName() {
+        return this.headerName;
+    }
+
+    public HeaderType getHeaderType() {
+        return this.headerType;
+    }
+
+    public boolean getValueAsBoolean() {
+        if (!(headerType == HeaderType.BooleanTrue || headerType == HeaderType.BooleanFalse)) {
+            throw new CrtRuntimeException("Invalid Event-stream header type");
+        }
+
+        return headerType == HeaderType.BooleanTrue;
+    }
+
+    private void setValue(boolean value) {
+        if (value) {
+            this.headerType = HeaderType.BooleanTrue;
+        } else {
+            this.headerType = HeaderType.BooleanFalse;
+        }
+    }
+
+    public byte getValueAsByte() {
+        checkType(HeaderType.Byte);
+        return headerValue[0];
+    }
+
+    private void setValue(byte value) {
+        headerType = HeaderType.Byte;
+        headerValue = new byte[] { value};
+    }
+
+    public short getValueAsShort() {
+        checkType(HeaderType.Int16);
```

Comment:
Confusing that we use the Java term "short" in the function name, but use the protocol term "int16" in the enum name.

We should pick one approach or the other. Always use the language-idiomatic term (ex: "short", "date", "bytes", etc), or always use the protocol term (ex: "int16", "timestamp", "byteBuf", etc).

I'm leaning towards calling protocol terminology everywhere. Like, the docs for a service will say "send an INT16 header named 'blah'", so it's nice if the same terminology is used in each language API.

Reply:
I think java devs would rather deal with type names mapped into their domain language. However the int* names refer to the protocol. I dunno. I don't care for it either, but it seems less bad than the alternative to me.